packages/nx/src/project-graph/utils/retrieve-workspace-files.ts

```diff
@@ -155,14 +186,25 @@ function buildAllWorkspaceFiles(
 function createProjectConfigurations(
   workspaceRoot: string,
   nxJson: NxJsonConfiguration,
-  configFiles: string[]
-): Record<string, ProjectConfiguration> {
+  configFiles: string[],
+  _ignorePluginInference = false
```

Comment:
Split this into multiple functions, one that does not use plugins, and this one which does